src/ci/init_repo.sh

```diff
@@ -41,26 +39,25 @@ function fetch_github_commit_archive {
     local cached="download-${module//\//-}.tar.gz"
     retry sh -c "rm -f $cached && \
         curl -f -sSL -o $cached $2"
-    mkdir $module
+    mkdir "$module"
     touch "$module/.git"
-    tar -C $module --strip-components=1 -xf $cached
-    rm $cached
+    tar -C "$module" --strip-components=1 -xf "$cached"
+    rm "$cached"
 }
 
 included="src/llvm-project src/doc/book src/doc/rust-by-example"
-modules="$(git config --file .gitmodules --get-regexp '\.path$' | cut -d' ' -f2)"
-modules=($modules)
+read -ra modules <<< "$(git config --file .gitmodules --get-regexp '\.path$' | cut -d' ' -f2)"
 use_git=""
-urls="$(git config --file .gitmodules --get-regexp '\.url$' | cut -d' ' -f2)"
-urls=($urls)
-for i in ${!modules[@]}; do
-    module=${modules[$i]}
+read -ra urls <<< "$(git config --file .gitmodules --get-regexp '\.url$' | cut -d' ' -f2)"
+
+for i in "${!modules[@]}"; do
```

Comment:
What is this doing? Why did you change it?

Reply:
This is the "correct" way to extract a space-separated list into an array, there can be some issues with globbing using ($...). But now that I look at it I can probably do without using `cut` and just using read directly.

Reply:
If I had to guess, this is what's causing CI to fail. I would revert the changes for now I think.